airflow/auth/managers/fab/security_manager/modules/db.py

```diff
@@ -257,10 +257,10 @@ def find_user(self, username=None, email=None):
                 return None
 
     def find_register_user(self, registration_hash):
-        return (
-            self.get_session.query(self.registeruser_model)
-            .filter(self.registeruser_model.registration_hash == registration_hash)
-            .scalar()
+        return self.get_session.scalar(
+            select(self.registeruser_mode)
+            .where(self.registeruser_model.registration_hash == registration_hash)
+            .limit(1)
         )
```

Comment:
This exposes a problem that might not be entirely in scope of the PR. If there is a hash cache, this lookup may return a wrong user and cause a security vulnarability. We should really either add a unique constraint to `registration_hash`, or use `one_or_none` here instead; both would remove the need to add a `limit`.

Reply:
Should I keep this changes or better revert it back?

Reply:
Keep it. We’re not sure how to change this code yet and adding this limit would be useful before we figure things out.